app/views/snapshots/_form.html.haml

```diff
@@ -72,6 +72,62 @@
         var drawToolStateHasChanged = function() {
           return originalData != drawingTool.canvas.toDataURL();
         };
+        var phone = iframePhone.getIFrameEndpoint();
+
+        // the drawing tool doesn't expose if the document is dirty but it does send events on undoable and redoable actions
+        // the first attempt at this kept a counter of undo events (decrementing when 'redo:possible' was received) but since
+        // the undo button itself sends a redo:possible event and there is no way to check if there are any other actions
+        // on the undo/redo stack, that counter was removed.  A simple flag is now set when anything undoable happens - this isn't
+        // ideal as if the user uses the undo button back to a "clean" document LARA will still think it is dirty
+        var listenForDrawingChanges = function() {
+          drawingTool.on('undo:possible', function () {
+            canClose = false;
+          })
+        };
+        var canClose = true;
+        var checkIfCanCloseFrom = function (from) {
+          if (canClose) {
+            phone.post('connected', false);
+            if (from === 'phoneHandler') {
+              phone.post('close')
+            }
+            return true;
+          }
+
+          $('<div style="padding: 10px; max-width: 500px; word-wrap: break-word;">You have unsaved changes.  Are you sure you want to close?</div>').dialog({
+              draggable: false,
+              modal: true,
+              resizable: false,
+              width: 'auto',
+              title: 'Confirm',
+              minHeight: 75,
+              buttons: {
+                Yes: function () {
+                  canClose = true;
+                  phone.post('disconnect');
```

Comment:
I don't see a handler for disconnect in LARA.  Is it necessary to disconnect in this case?
If connected LARA will send the canClose message. And then the labbook will immediately respond with 'close'.